src/pages/ServiceGraph/SummaryPanelGraph.tsx

```diff
@@ -49,28 +49,28 @@ export default class SummaryPanelGraph extends React.Component<SummaryPanelPropT
 
   render() {
     const cy = this.props.data.summaryTarget;
+    if (!cy) {
+      return null;
+    }
+
     let numNodes = cy
-      ? cy
-          .nodes()
-          .filter('[!groupBy]')
-          .size()
-      : 0;
-    let numEdges = cy ? cy.edges().size() : 0;
+      .nodes()
+      .filter('[!groupBy]')
+      .size();
+    let numEdges = cy.edges().size();
     let rate = 0;
     let rate3xx = 0;
     let rate4xx = 0;
     let rate5xx = 0;
     let safeRate = (s: string) => {
       return s === undefined ? 0.0 : parseFloat(s);
     };
-    if (cy) {
-      cy.edges().forEach(edge => {
-        rate += +safeRate(edge.data('rate'));
-        rate3xx += +safeRate(edge.data('rate3XX'));
-        rate4xx += +safeRate(edge.data('rate4XX'));
-        rate5xx += +safeRate(edge.data('rate5XX'));
-      });
-    }
+    cy.edges().forEach(edge => {
```

Comment:
Try not to use `forEach` in the future, it forces you to use mutable data structures. Most of the time `map` or `reduce` is a better option because its immutable. With React/Redux now, immutability is a big issue.

Reply:
OK.  This was existing code, but I was the original author.  I'll look at the alternatives.